csrc/pf_inner.c

```diff
@@ -1106,6 +1106,15 @@ DBUG(("XX ah,m,l = 0x%8x,%8x,%8x - qh,l = 0x%8x,%8x\n", ah,am,al, qh,ql ));
 	    }
 	    endcase;
 
+	case ID_FILE_RESIZE: /* ( ud fileid -- ior ) */
+	    {
+		FileStream *File = (FileStream *) TOS;
+		ucell_t SizeHi = (ucell_t) M_POP;
+		ucell_t SizeLo = (ucell_t) M_POP;
+		TOS = sdResizeFile( File, SizeLo, SizeHi );
```

Comment:
On 64-bit Linux, off_t is 64 bits and long is also 64 bits. On 32-bit Linux, long is 32 bits; off_t is 32 bits by default but can be changed to 64 by defining  _FILE_OFFSET_BITS=64.  On Windows, long is always 32 bits even on "64-bit Windows"; I don't know what off_t is there.

A real problem is that off_f, fseeko, and fseeko aren't part of ANSI C.  I think they should not be used
by default.

GCC has a __int128 type on some machines, but not everywhere (Alpha?).  And while there is a int64_t in C99 (but not in C89), threre is no int128_t in C99. So I think the only portable way to pass double cell integers is either a Lo/Hi pair or a struct with Lo and Hi fields.